stac_fastapi/pgstac/transactions.py

```diff
@@ -162,7 +162,10 @@ async def update_collection(
         **kwargs,
     ) -> Optional[Union[stac_types.Collection, Response]]:
         """Update collection."""
-        col = collection.model_dump(mode="json")
+
+        # this is hacky - no one needs to tell me - some test pass with this :)
+        col = collection if "id" in collection else collection.model_dump(mode="json")
+        # col = collection.model_dump(mode="json")
```

Comment:
can you explain this?

Reply:
these are the errors:

FAILED tests/clients/test_postgres.py::test_update_collection[api_client3] - AttributeError: 'dict' object has no attribute 'model_dump'
FAILED tests/resources/test_collection.py::test_update_collection[api_client3] - AttributeError: 'dict' object has no attribute 'model_dump'
FAILED tests/resources/test_collection.py::test_update_new_collection[api_client3] - AttributeError: 'dict' object has no attribute 'model_dump'
FAILED tests/resources/test_item.py::test_update_collection[api_client3] - AttributeError: 'dict' object has no attribute 'model_dump'

Reply:
I'm not sure why a dict is being passed but only in the update collection route? I will look more into it when I get a chance.

Reply:
Maybe when this was added? https://github.com/stac-utils/stac-fastapi/blob/e4e4120ae6d1a9fe80910073af5889ae08c418f8/stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py#L32-L36

Reply:
It doesn't seem to make sense but I don't know what else would have changed that would only cause this error with the PUT collections route?

Reply:
Actually this makes no sense because this change was pre 2.5.5